js/src/model.rs

```diff
@@ -10,6 +10,85 @@ thread_local! {
     pub static FROM_JS: FromJsConverter = FromJsConverter::default();
 }
 
+#[wasm_bindgen(typescript_custom_section)]
+const TYPESCRIPT_CUSTOM_SECTION: &str = r###"
+export function literal(value: string | undefined, languageOrDataType?: string | NamedNode): Literal;
+export function quad(subject: Quad_Subject, predicate: Quad_Predicate, object: Quad_Object, graph: Quad_Graph): Quad;
+
+export class BlankNode {
+    readonly termType: "BlankNode";
+    readonly value: string;
+
+    equals(other: Term | null | undefined): boolean;
+    toString(): string;
+}
+
+export class DefaultGraph {
+    readonly termType: "DefaultGraph";
+    readonly value: "";
+
+    equals(other: Term | null | undefined): boolean;
+    toString(): string;
+}
+
+export class Literal {
+    readonly datatype: NamedNode;
+    readonly language: string;
+    readonly termType: "Literal";
+    readonly value: string;
+
+    equals(other: Term | null | undefined): boolean;
+    toString(): string;
+}
+
+export class NamedNode {
+    readonly termType: "NamedNode";
+    readonly value: string;
+
+    equals(other: Term | null | undefined): boolean;
+    toString(): string;
+}
+
+export interface BaseQuad {
+    readonly graph: Term;
+    readonly object: Term;
+    readonly predicate: Term;
+    readonly subject: Term;
+    readonly termType: "Quad";
+    readonly value: "";
+
+    equals(other: Term | null | undefined): boolean;
+    toString(): string;
+}
```

Comment:
I left it out initially, but:
* `DatasetCore` `match` method takes `Term` parameters
* The RDF/JS `Term` unions `BaseQuad` instead of `Quad`

So there's type incompatibility if `match` takes `Quad` instead of `BaseQuad`.
I admit I don't really understand the rationale behind that, and it doesn't seem to be documented in sources I could find.

I copied `BaseQuad`, the `Term` union, and the `match` signature from the RDF/JS types.

Reply:
Ok! Make sense. Thank you! Maybe add a comment?

Reply:
I added comments to the generated TypeScript and a headline comment to the Rust.